ESLintBear: Add filename context

oreqizer opened the pull request:
Adds information about current file location for ESLint.
Aids in relative imports.

Fixes https://github.com/coala-analyzer/coala-bears/issues/741

underyx commented:
💥  🎰  💥

underyx commented on bears/js/ESLintBear.py:
```diff
@@ -35,7 +35,11 @@ def create_arguments(filename, file, config_file,
+               '--no-color',\
+               '-f=json',\
+               '--stdin',\
+               '--stdin-filename=' + filename
```
Backslashes 😧 

I wonder why this isn't checked by static analysis checks, using backslashes to escape newlines is considered a pretty bad practice. However!

1. `args = 1, 2, 3` is just syntactic sugar for `args = (1, 2, 3)` (which is creating a `tuple` and is therefore syntactic sugar for `args = tuple(1, 2, 3)`)
2. Newlines don't need to be escaped inside parentheses.

Which means that a multiline
```
args = 1,\
       2,\
       3
```
can also be written as
```
args = (
    1,
    2,
    3,
)
```

oreqizer commented:
@underyx Fixed. Thanks for the heads up! I'm not really familiar with all of Python syntax and semantics yet.

sils1297 commented:
Hi,

it looks like your PR is out of date and needs a rebase. [This page](http://coala.readthedocs.org/en/latest/Users/Tutorials/Git_Help.html) may help you to get started on this. (*Do not press the update branch button. It's evil.*)

Add the original coala repository as a remote:

```
git remote add upstream https://github.com/coala-analyzer/coala.git
```

Then go to your branch and rebase it onto our master:

```
git checkout <yourbranch>
git fetch upstream
git rebase --interactive upstream/master
```

Now push it up. You will need to do a force push because you change history here (do not do this on master branches :)):

```
git push --force
```

If you're just looking to fix an issue very quickly and not interested in contributing to coala long term, we can fix it up for you - just tell us :).

Ping us on [our gitter channel](https://gitter.im/coala-analyzer/coala) if you need help!

sils1297 commented:
also can we maybe have a test for this? I feel like we defiintely should have one

underyx commented:
Thanks! That's alright @oreqizer, that's why we're here :D

sils1297 commented:
ack https://github.com/coala-analyzer/coala-bears/pull/777/commits/613afd9f9c68282f4eda0306c23590351ca51c64

underyx commented on bears/js/ESLintBear.py:
```diff
@@ -17,7 +17,9 @@ class ESLintBear:
-    REQUIREMENTS = {NpmRequirement('eslint', '2')}
+    REQUIREMENTS = {NpmRequirement('babel-eslint', '6'),
+                    NpmRequirement('eslint', '1'),
+                    NpmRequirement('eslint-plugin-import', '2')}
```
How come these versions don't match the `package.json` ones?

oreqizer replied:
ohh you're right. i'll fix

sils1297 commented:
argh sorry, @AbdealiJK did a lot of bugfixes today and also touched parts of eslintbear which conflicts with your stuff so you'll have to rebase and resolve them. Tell us if you need aything. Sorry :/

sils1297 commented:
I'll pass you a few comments in the meantime

sils1297 commented on bears/js/ESLintBear.py:
```diff
@@ -47,7 +47,12 @@ def create_arguments(filename, file, config_file,
+        return """
+        {
+            "extends": "eslint:recommended",
+            "parser": "babel-eslint",
```
I think somebody raised a concern on wether we want this to be default or not but if you think it's safe I'd go with it. Still I'd like you to confirm that once again.

oreqizer replied:
It *is* the most common practice, but I am also very hesitant on making it default. I'll revert it during the rebasing just to keep stuff simple.

sils1297 replied:
ok so we probably want to add configurability later and make it a default

sils1297 commented:
your code looks good if the conflicts wouldn't occur I'd merge it right ahead. Since this is a bugfix I'll wait with the coala brears 0.8.3 release until we merge this :)

sils1297 commented:
marking WIP because of the conflicts though you shouldn't need to change anything

oreqizer commented:
I removed a commit that wasn't a good idea, and squashed the remaining two together, as:

* one was useless without the other
* second one didn't work without the first

so I think they make sense being together.

sils1297 commented:
@oreqizer github still shows conflicts, I think that's why the build didn't start :/

oreqizer commented:
oh, forgot to fetch upstream.. on it!

sils1297 commented:
ack https://github.com/coala-analyzer/coala-bears/pull/777/commits/96c86cd75a4b97357cc8f28c2c65796476436e21

sils1297 commented:
@rultor merge

rultor commented:
> @rultor merge

@sils1297 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/9528-244727638)

rultor commented:
> @rultor merge

@sils1297 Done! FYI, the full log is [here](http://www.rultor.com/t/9528-244727638) (took me 1min)
